flyteidl/protos/flyteidl/admin/agent.proto

```diff
@@ -90,14 +123,18 @@ message Resource {
   repeated core.TaskLog log_links = 4;
   // The phase of the execution is used to determine the phase of the plugin's execution.
   core.TaskExecution.Phase phase = 5;
+  // Custom data specific to the agent.
+  google.protobuf.Struct custom_info = 6;
 }
 
 // A message used to delete a task.
 message DeleteTaskRequest {
   // A predefined yet extensible Task type identifier.
-  string task_type = 1;
+  string deprecated_task_type = 1 [deprecated = true];
```

Comment:
cc @eapolinario CI failed to build the `flyte-kit/docs` because I renamed this field. I've updated the agent in this flytekit [PR](https://github.com/flyteorg/flytekit/pull/2146/files#diff-9f7af27264f8773b069e8200804c224fe19a6fcaaf9dc33edc644f5351cbb3beR123-R126).

flytekit PR needs to be merged first to fix the CI error, but this PR must be merged before merging flytekit PR since we need to cut a beta release for flytekit. there is a deadlock...